pkg/agent/run.go

```diff
@@ -247,8 +254,21 @@ func getConfiguration() (Config, client.Client) {
 	var preflightClient client.Client
 	switch {
 	case credentials != nil:
-		log.Println("A credentials file was specified, using oauth authentication.")
-		preflightClient, err = client.NewOAuthClient(agentMetadata, credentials, baseURL)
+		switch creds := credentials.(type) {
```

Comment:
Suggestion: can we have the nested switch in a separate function?

Reply:
Yes, moved.